widget/embedded/src/utils/ui.ts

```diff
@@ -28,47 +28,53 @@ export function customizedThemeTokens(
 ): CustomizedTheme {
   const baseColors = baseThemeTokens.colors;
 
-  const darkColorsWithDefaults = generateColors(
-    {
-      ...baseColors,
-      ...defaultDarkColors,
-    },
-    true,
-    colors?.dark
-  );
-  const lightColorsWithDefaults = generateColors(
-    baseColors,
-    false,
-    colors?.light
-  );
-  const hasDefaultDarkColors = Object.keys(darkColorsWithDefaults).length > 0;
-  const hasDefaultLightColors = Object.keys(lightColorsWithDefaults).length > 0;
-
-  let light: CustomizedTheme['light'] = undefined;
-  let dark: CustomizedTheme['dark'] = undefined;
-
-  if (hasDefaultLightColors) {
-    const tokens = { colors: lightColorsWithDefaults };
+  const baseDarkColors = {
+    ...baseColors,
+    ...defaultDarkColors,
+  };
+
+  const darkColorChanges = !!colors?.dark
```

Comment:
I suggest to change the code structure to something like this (I only do for `light`, `dark` will be same.).

```ts
if(color.light){

   const expandColors = findChangedColors(baseColors, colors.light);   
   if(expandColors){
      const lightColors = expandedThemeColors(baseColors, expandColors);
      const tokens = { colors: lightColors };
      // ...
   }
   
}
```